src/adapter/src/coord/privatelink_status.rs

```diff
@@ -57,36 +48,18 @@ impl Coordinator {
                         // to continue to work, despite not being polled
                         rate_limiter.until_ready().await;
 
-                        // Events to be written, de-duped by connection_id
-                        let mut event_map = BTreeMap::new();
-
-                        for event in new_events {
-                            match last_written_events.get(&event.connection_id) {
-                                // Ignore if an event with this time was already written
-                                Some(time) if time >= &event.time => {}
-                                _ => {
-                                    last_written_events
-                                        .insert(event.connection_id.clone(), event.time.clone());
-                                    event_map.insert(event.connection_id, event);
-                                }
-                            }
-                        }
-
-                        // Send the event batch to the coordinator to be written
+                        // Fine if receiver hangs up
                         let _ =
-                            internal_cmd_tx.send(Message::PrivateLinkVpcEndpointEvents(event_map));
+                            internal_cmd_tx.send(Message::PrivateLinkVpcEndpointEvents(new_events));
                     }
                 }
             });
         }
     }
 
-    pub(crate) async fn write_privatelink_status_updates(
-        &mut self,
-        events: BTreeMap<GlobalId, VpcEndpointEvent>,
-    ) {
+    pub(crate) async fn write_privatelink_status_updates(&mut self, events: Vec<VpcEndpointEvent>) {
```

Comment:
Is there a way to preserve the de-duping of repeated status updates using the state now stored in `self.controller.storage.collection_status_manage.previous_statuses` ? Or is it the plan to dedupe across all status updates at some point?

Reply:
I did not realize these updates were not going through the `CollectionStatusManager`! Will see if I can tie those together in just a bit or else will revert the commit that made those changes.

Reply:
Just reverted this change--tysm for catching that I had assumed this was wired together improperly. I had not realized we now have multiple `CollectionManager`s.